distributed/utils_test.py

```diff
@@ -615,6 +613,15 @@ def security():
     return tls_only_security()
 
 
+def _close_queue(q):
+    q.close()
```

Comment:
There were explicit calls to close the internal pipes of the queue. anybody has a clue why this was in here? I prefer not to and rely on the public close

Reply:
the plain `q.close()` goes on to close the *read* end of the pipe, but not the *write* end: https://github.com/python/cpython/blob/3.8/Lib/multiprocessing/queues.py#L137

if you're going to just use one method call here - you can shorten the code to `stack.callback(queue.close)`

Reply:
first introduced here https://github.com/dask/distributed/pull/1317/files#diff-454ff0342d1c0724fc6e164fcb884d6727676fd560869fab3331a11adc812d56R412-R413

Reply:
Well, the writer is closed in a very awkward way 
A finalizer puts a sentinel in a buffer https://github.com/python/cpython/blob/b1bc04df817ecb3fb9fdbefa3782efea54488e6f/Lib/multiprocessing/queues.py#L184-L188

which the control loop picks up and closes the writer

https://github.com/python/cpython/blob/b1bc04df817ecb3fb9fdbefa3782efea54488e6f/Lib/multiprocessing/queues.py#L231-L236

Reply:
ah but if you don't `q.put(` anything then the thread isn't started and so the q._writer is leaked